src/components/DateInput.jsx

```diff
@@ -232,6 +235,7 @@ class DateInput extends React.PureComponent {
           readOnly={typeof readOnly === 'boolean' ? readOnly : isTouch}
           required={required}
           aria-describedby={screenReaderMessage && screenReaderMessageId}
+          aria-labelledby={ariaLabelledBy}
```

Comment:
You may also want to explicitly not set `aria-label` if `aria-labelledby` is passed in so that there isn't a potential conflict on which one takes precedence.